steward/src/commands/start.rs

```diff
@@ -20,6 +29,10 @@ impl Runnable for StartCmd {
         let config = APP.config();
         info!("Starting application");
         abscissa_tokio::run(&APP, async {
+            // currently allows the threads to detach since we aren't capturing the JoinHandle
```

Comment:
Given that there is no join call, do these shut down cleanly?

Reply:
Do you mean ends gracefully or ends at all? My understanding is it is killed ungracefully when the process exits.

Reply:
Yeah I guess the process itself doesn't really have a graceful shutdown as it is. Just trying to reason about how the threads are spawned and detached.

Reply:
One thing we may want is some keep-alive functionality which implies joining these threads so that we can respawn them on error.

Reply:
Join would only complete after all the threads have exited, no? We're currently effectively blocking on `serve` so in theory you could join all three, but I'm not sure that'd fix the restart issue. Plus, the only way for these threads to exit right now is with a panic, right? I think all errors are handled.

Reply:
Yeah all errors are handled, the validator would just have to restart Steward if they weren't submitting proposal corks due to a panic. I haven't looked at what tokio offers by way of keepalive but I guess it would be it's own detached thread that internally has joined threads?

Reply:
You'd probably have to structure it as a subtask or something, yeah. The weird thing though is that if something in the thread's processing is instigating a panic, restarting it is unlikely to fix the issue. It almost feels like the more reasonable thing to do is crash the entire process, because then maybe the operator would see that and devs can patch the issue. Failing "quietly" and logging is an option, but might not be noticed. This is also assuming that a crashing process would be noticed and reported though, and not just restarted automatically. Otherwise you basically are in the position of waiting for a human (or some bot process) to notice something it expected to happen didn't happen. For an approved cellar cache issue, that's probably relatively straightforward, but very annoying for a governance prop.

Reply:
In a more traditional setting this would fire an alert in Sentry or something, but we don't support that sort of thing at the moment and validator setups are inconsistent. Might be worth looking into adding an option for exception handling integration (here and elsewhere).